### ISSUE TITLE
Enabling colemak keybinds breaks evil-respect-visual-line-mode.

### ISSUE THREAD

**Issue by NiamhFerns** [wbolster/emacs-evil-colemak-basics#16]
When using visual line numbers, you need to enable `evil-respect-visual-line-mode` (before evil loads) and `visual-line-mode` so that line jump commands are actually respected. 

See [here](https://www.reddit.com/r/emacs/comments/r3ahhm/how_to_emulate_vim_relative_line_numbers_with/) for detailed write up on what I mean.

The issue is, when I enable colemak mode, it no longer respects the the visual line mode. (Skips over visual lines and moves the cursor based on logical lines). This completely breaks the line number based movement commands.

Say I am on line 1 and line numbers show the line I want to be on is line 25 (with visual line numbers enabled) but there are 5 lines of folded text between me and the line I want to be on. When I type '25n' I end up on line 29 instead of line 25. This defeats the entire purpose of having relative/visual line numbers.

Is there maybe something I'm missing when setting the relevant variables and packages in my config?

At the top of my init.el file I have `(setq evil-respect-visual-line-mode 't)`, in my packages.el I have `(package! evil-colemak-basic)` which includes the relavant MELPA package and then in my config.el I have
```
(use-package evil-colemak-basics
    :init
        (setq evil-colemak-basics-layout-mod `mod-dh)
    :config
        (global-evil-colemak-basics-mode))
```
This is with DOOM Emacs but afaik this should apply to vanilla Emacs as well.

**Comment by wbolster** [comment]
which commands are you using exactly? there's `evil-next-line` (`n`) and `evil-next-visual-line` (`gn`).

it seems the latter works fine with relative line numbering, while the former does not. it seems this happens both with and without this package activated.

**Comment by KaosBunny** [comment]
When I am using QWERTY, `gj` is bound to `evil-next-visual-line` and `j` is bound to `evil-next-line`. When I enable word wrapping (`M-x visual-line-mode`) These 2 get switched around so that `j` is bound to `evil-next-visual-line` and visa versa.

When using colemak though, this switch doesn't happen. `evil-next-visual-line` is still bound to `gn` instead of `n`.

**Comment by wbolster** [comment]
aha, do you happen to know how evil achieves this?

(i have something very similar in my dotfiles for years but it's rather hacky)

**Comment by KaosBunny** [comment]
I'm not really sure how it's done under the hood (still very new to this sort of thing) but there is a variable you can set before evil loads that will tell it to switch keybindings when it detects that `visual-line-mode` has been enabled. (A hook I'm assuming, but again, I'm really not sure.)

I have it enabled at the top of my init.el file with `(setq evil-respect-visual-line-mode 't)`.

**Comment by wbolster** [comment]
looks like this part of `evil-integration.el` is responsible for it:

``` elisp
(when evil-respect-visual-line-mode
  ;; ...
  (evil-define-minor-mode-key 'motion 'visual-line-mode
    "j" 'evil-next-visual-line
    "gj" 'evil-next-line
    "k" 'evil-previous-visual-line
    "gk" 'evil-previous-line
    "0" 'evil-digit-argument-or-evil-beginning-of-visual-line
    "g0" 'evil-beginning-of-line
    "$" 'evil-end-of-visual-line
    "g$" 'evil-end-of-line
    "V" 'evil-visual-screen-line))
```

that's quite annoying tbh since it's not dynamic, and defined outside the affected commands :(

**Comment by KaosBunny** [comment]
Ah, yeah that's frustrating. :c 

I've run into another issue, as well (although not strictly related to this). For some reason (at least on my laptop), the `evil-next-visual-line` and `evil-previous-visual-line` commands make scrolling through text incredibly laggy when held down. Much more so than is normal for Emacs with line numbers enabled.

**Comment by wbolster** [comment]
i guess what you describe in https://github.com/wbolster/emacs-evil-colemak-basics/issues/16#issuecomment-982417534 is unrelated to this package? upstream evil is a better place for that

**Comment by mk3z** [comment]
Solved in #19
